File: solutions/go/clock/1/clock.go
package clock

import "fmt"

// Clock represents a time without a date
type Clock struct {
	hour   int
	minute int
}

// normalize ensures that hour and minute are within proper ranges
func normalize(h, m int) (int, int) {
	totalMinutes := h*60 + m

	// wrap around 24 hours = 1440 minutes
	totalMinutes = ((totalMinutes % 1440) + 1440) % 1440

	h = totalMinutes / 60
	m = totalMinutes % 60

	return h, m
}

// New creates a new Clock
func New(h, m int) Clock {
	h, m = normalize(h, m)
	return Clock{hour: h, minute: m}
}

// Add adds minutes to the clock
func (c Clock) Add(m int) Clock {
	return New(c.hour, c.minute+m)
}

// Subtract subtracts minutes from the clock
func (c Clock) Subtract(m int) Clock {
	return New(c.hour, c.minute-m)
}

// String formats the clock as HH:MM
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

File: solutions/typescript/clock/1/clock.ts
export class Clock {

  private minutes : number;
  
  constructor(hour: number, minute: number = 0) {
    this.minutes = ((hour * 60 + minute) % (24 * 60) + (24 * 60)) % (24 * 60)
  }

  public toString(): string {
    const hours = Math.floor(this.minutes / 60)
    const mins = this.minutes % 60
    return `${hours.toString().padStart(2, '0')}:${mins.toString().padStart(2, '0')}`
  }

  public plus(minutes: number): Clock {
    return new Clock(0, this.minutes + minutes);
  }

  public minus(minutes: number): Clock {
    return new Clock(0, this.minutes - minutes);
  }

  public equals(other: Clock): boolean {
    return this.minutes === other.minutes
  }
}
